app/controllers/users_controller.rb

```diff
@@ -13,16 +21,34 @@ def new
   end
 
   def create
-    user = User.new(user_params)
-    
-    authenticate!(:password) if user.save
-    after_login_redirect
+    @user = User.new(user_params)
+    redirect_url = nil
+
+    if @user.save
+      authenticate!(:password_unconfirmed)
+      @user.send_confirmation
+      redirect_url = accept_invite_url(params[:token]) if params[:token].present?
+      after_login_redirect(redirect_url)
+    else
+      render :new
```

Comment:
You need to add specs so we check what happens when the user can't be saved. For example test when you pass a short password and check that we render new.